Assets/MixedRealityToolkit/Services/MixedRealityToolkit.cs

```diff
@@ -121,23 +121,191 @@ public void ResetConfiguration(MixedRealityToolkitConfigurationProfile profile)
         /// <remarks>
         /// Systems can only be registered once by <see cref="Type"/>
         /// </remarks>
-        public static IReadOnlyDictionary<Type, IMixedRealityService> ActiveSystems => activeSystems;
+        public IReadOnlyDictionary<Type, IMixedRealityService> ActiveSystems => new Dictionary<Type, IMixedRealityService>(activeSystems) as IReadOnlyDictionary<Type, IMixedRealityService>;
 
         private static readonly List<Tuple<Type, IMixedRealityService>> registeredMixedRealityServices = new List<Tuple<Type, IMixedRealityService>>();
 
         /// <summary>
         /// Local service registry for the Mixed Reality Toolkit, to allow runtime use of the <see cref="Microsoft.MixedReality.Toolkit.Core.Interfaces.IMixedRealityService"/>.
         /// </summary>
-        public static IReadOnlyList<Tuple<Type, IMixedRealityService>> RegisteredMixedRealityServices => registeredMixedRealityServices;
-
-        /// <summary>
-        /// Local service registry for the Mixed Reality Toolkit, to allow runtime use of the <see cref="Microsoft.MixedReality.Toolkit.Core.Interfaces.IMixedRealityService"/>.
-        /// </summary>
-        [Obsolete("Use RegisteredMixedRealityServices instead.")]
-        public List<Tuple<Type, IMixedRealityExtensionService>> MixedRealityComponents => null;
+        public IReadOnlyList<Tuple<Type, IMixedRealityService>> RegisteredMixedRealityServices => new List<Tuple<Type, IMixedRealityService>>(registeredMixedRealityServices) as IReadOnlyList<Tuple<Type, IMixedRealityService>>;
 
         #endregion Mixed Reality runtime service registry
 
+        #region IMixedRealityServiceRegistrar implementation
+
+        /// <inheritdoc />
+        public bool RegisterService<T>(T serviceInstance) where T : IMixedRealityService
+        {
+            return RegisterServiceInternal<T>(serviceInstance);
+        }
+
+        /// <inheritdoc />
+        public bool RegisterService<T>(
+            Type concreteType, 
+            SupportedPlatforms supportedPlatforms = (SupportedPlatforms)(-1), 
+            params object[] args) where T : IMixedRealityService
+        {
+            if (isApplicationQuitting)
+            {
+                return false;
+            }
+
+#if !UNITY_EDITOR
```

Comment:
Future comment for us, for these it might be helpful to add to our Extensions or have a helper class abstract this out. I find that having interspersed #if defs can be tricky visually, when really what the code should read is like:

if (IsPlatformSupported(supportedPlatforms))

And the helper function could abstract away the platform specific issues (i.e. as a caller in RegisterService I really don't care about how to see this check is done, only that it is handled on my behalf.

Reply:
So yeah I don't think we need to make a change in this persay, I just want to see if others agree that the approach described above is agreeable to them.

Reply:
great thing to add in the future. would be additive not a breaking change :)